### PR TITLE
Feat/2707 add product

### PR DESCRIPTION
**Pull request by GuilhermeMota93**
## RESOURCES
[The api definition](https://woocommerce.github.io/woocommerce-rest-api-docs/#create-a-product)

## CHANGES / DESCRIPTION

This PR will contain the work to allow a remote communication for the **creation of a new product**. This chunk of work meant:
- add new product actions in `WCProductAction`
- handle of new actions and new event `OnProductCreated` in `WCProductStore`
- add new product function in `ProductRestClient`
- added tests to the new request in **APITesting_WCProduct**, **ReleaseStack_WCProductTest** and **WCProductStoreTest.kt**

Following the `README`, I added the initial action in the `WCProductAction`; I called them:
- **ADD_NEW_PRODUCT** 
- **ADDED_NEW_PRODUCT**

In the `WCProductStore`, the `onAction` method will be listening to these actions that, upon trigger will behave as such:

**1)** `ADD_NEW_PRODUCT`
Will call the `createNewProduct` of the `ProductRestClient` that will make the remote connection and respond with success/error, mapping the response model accordingly. This done, it will trigger the correct action for completion, with the data;

**2)** `ADDED_NEW_PRODUCT`
Will call the `handleAddNewProduct`, from a previous completion of a request, and create the success/error scenarios accordingly, using the `OnProductCreated` for proper communication of the result.

This will get emitted to whoever is listening and trigger the UI updates (example app has this integration 👍 also, in screenshots bellow)

Let me know your thoughts on it 🙏 

## SCREENSHOTS
Product created
-
 

New Product fetched
-

Tests
-
Screenshot 2020-08-25 at 23 56 17

### CODE REVIEW

**Comment by GuilhermeMota93** [comment]
The Mocked Connected Tests are failing though 😬 

Couldn't get the `APITesting_WCProduct` nor the `ReleaseStack_WCProductTest` running as I got the error: 
```
TEST_WC_PRODUCT_ID - NumberFormatException: For input string: "FIXME"
```

File: example/src/test/java/org/wordpress/android/fluxc/wc/product/WCProductStoreTest.kt [hunk 1]
```diff
@@ -209,4 +211,26 @@ class WCProductStoreTest {
...
+            name = "test new product"
+            description = "test new description"
+        }
+        val site = SiteModel().apply { id = productModel.localSiteId }
+
+        // when
+        ProductSqlUtils.insertOrUpdateProduct(productModel)
+        val payload = RemoteAddNewProductPayload(site, productModel)
+        productStore.onAction(WCProductActionBuilder.newAddedNewProductAction(payload))
+
```
Comment (GuilhermeMota93) [inline example/src/test/java/org/wordpress/android/fluxc/wc/product/WCProductStoreTest.kt:228]: Having second thoughts about this naming `newAddedNewProductAction` here. It feels a bit confusing "new-added-new".

I might change the actions to be simpler, what are your thoughts @0nko ?

File: plugins/woocommerce/src/main/java/org/wordpress/android/fluxc/action/WCProductAction.java [hunk 2]
```diff
@@ -128,5 +132,7 @@
     @Action(payloadType = RemoteProductTagsPayload.class)
     FETCHED_PRODUCT_TAGS,
     @Action(payloadType = RemoteAddProductTagsResponsePayload.class)
-    ADDED_PRODUCT_TAGS
+    ADDED_PRODUCT_TAGS,
+    @Action(payloadType = RemoteAddNewProductPayload.class)
+    ADDED_NEW_PRODUCT
```
Comment (GuilhermeMota93) [inline plugins/woocommerce/src/main/java/org/wordpress/android/fluxc/action/WCProductAction.java]: As mentioned above, maybe the naming could be simpler, something like `ADDED_PRODUCT`.

File: plugins/woocommerce/src/main/java/org/wordpress/android/fluxc/action/WCProductAction.java [hunk 3]
```diff
@@ -86,6 +88,8 @@
     FETCH_PRODUCT_TAGS,
     @Action(payloadType = AddProductTagsPayload.class)
     ADD_PRODUCT_TAGS,
+    @Action(payloadType = AddNewProductPayload.class)
+    ADD_NEW_PRODUCT,
```
Comment (GuilhermeMota93) [inline plugins/woocommerce/src/main/java/org/wordpress/android/fluxc/action/WCProductAction.java]: Thinking of simplifying this name here. Reason being, the generated action builder for it looks odd (imo). 

I tried following the `README` recommendation here, but maybe `ADD_PRODUCT`, a simpler naming, would be best?

Comment (0nko) [reply plugins/woocommerce/src/main/java/org/wordpress/android/fluxc/action/WCProductAction.java]: `ADD_PRODUCT` is clear enough 👍

File: example/src/test/java/org/wordpress/android/fluxc/wc/product/WCProductStoreTest.kt (hunk 1 above)
Comment (0nko) [reply example/src/test/java/org/wordpress/android/fluxc/wc/product/WCProductStoreTest.kt:228]: Yeah, I agree, I'd remove the "new" here.

**Review by 0nko — COMMENTED** [review]
This looks good 👍. I left some comments in the code. 

Don't worry about the API tests for now because you don't have the credentials to run them. If you can get the mocked tests and unit tests working then it's good.

**Comment by GuilhermeMota93** [comment]
Cool 👍 will update that naming and push the changes later today. Thank you!

**Pull request merged**